jbmc/src/janalyzer/janalyzer_parse_options.cpp

```diff
@@ -358,24 +362,77 @@ int janalyzer_parse_optionst::doit()
 
   register_languages();
 
-  if(cmdline.args.size() > 1)
+  if(!((cmdline.isset("jar") && cmdline.args.size() == 0) ||
```

Comment:
Hm, I actually prefer `not(or_i good_case_i)` to `and_i not(good_case_i)`

Reply:
I raised this, because the reader needs to 1) figure out matching parentheses and 2) read all the lines at once. Maybe you just want to introduce some named subexpressions?